pkg/router/skipper.go

```diff
@@ -0,0 +1,202 @@
+package router
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+
+	flaggerv1 "github.com/weaveworks/flagger/pkg/apis/flagger/v1beta1"
+)
+
+/*
+Skipper Principles:
+
+* if only one backend has a weight, only one backend will get 100% traffic
+* if two of three backends has a weight, only two should get traffic.
+* if two backends doesn't have any weight, they get equal amount of traffic.
```

Comment:
This (3rd) one was long under discussion and I don't know if this would change in the future, but I would rather say `it's undefined and right now they get equal amount of traffic`.

Reply:
thanks - implemented https://github.com/o11n/flagger/pull/5